sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/V2ExpressionUtils.scala

```diff
@@ -159,7 +159,7 @@ object V2ExpressionUtils extends SQLConfHelper with Logging {
         StaticInvoke(scalarFunc.getClass, scalarFunc.resultType(),
           MAGIC_METHOD_NAME, arguments, inputTypes = declaredInputTypes,
           propagateNull = false, returnNullable = scalarFunc.isResultNullable,
-          isDeterministic = scalarFunc.isDeterministic)
+          isDeterministic = scalarFunc.isDeterministic, scalarFunction = Some(scalarFunc))
```

Comment:
Because spark already construct the StaticInvoke by calling these methods. e.g. `scalarFunc.getClass`, `scalarFunc.resultType()` and so on.
Please keep the same way.